protoc-gen-gogo/generator/generator.go

```diff
@@ -1183,12 +1183,26 @@ func (g *Generator) generateHeader() {
 // The path is a comma-separated list of integers.
 // See descriptor.proto for its format.
 func (g *Generator) PrintComments(path string) {
-	if loc, ok := g.file.comments[path]; ok {
-		text := strings.TrimSuffix(loc.GetLeadingComments(), "\n")
-		for _, line := range strings.Split(text, "\n") {
-			g.P("// ", strings.TrimPrefix(line, " "))
-		}
+	text := g.Comments(path)
```

Comment:
I think its import that the ok is accessible here.
I don't mind a little code duplication.
Copy the Comments method as is to helper.go file, because I think it actually has a nice API, but the PrintComments method requires the ok.
Revert the changes to PrintComments and leave it as it was.
I think this gives us the best of both worlds, what do you think?